web/api/queries/query.c

```diff
@@ -970,13 +970,6 @@ static RRDR *rrd2rrdr_fixedstep(
 
     // we align the request on requested_before
     time_t before_wanted = before_requested;
-    if(likely(before_wanted > last_entry_t)) {
-        #ifdef NETDATA_INTERNAL_CHECKS
-        error("INTERNAL ERROR: rrd2rrdr() on %s, before_wanted is after db max", st->name);
-        #endif
-
-        before_wanted = last_entry_t - (last_entry_t % ( ((aligned)?group:1) * update_every ));
-    }
```

Comment:
A question, not sure: Could this affect health?...

Reply:
It could affect it I think... But maybe in a better way.

In normal operation I'm observing that before_wanted = last_entry_t.

If by chance (e.g. high load on the host) what health requests as before_wanted is > than last_entry_t (especially in 5s updated charts), the extra null value looks like it is considered in the calculation.

In a test alert, that sums a constant chart for the last minute, health seems to get a different value back when before_wanted = last_entry_t than when before_wanted > last_entry_t.

But: When I re-introduced the removed code from this PR, it seems I get errors when before_wanted > last_entry_t... Not sure why, need to dig more...

I don't think it's a blocker for this PR, but something we need I think to look more into.